Sequence.java

```diff
@@ -1,26 +1,30 @@
-import java.util.Scanner;
-
+/** 
+ * Prints the type of sequence:non decreasing or not. 
+ */
 public class Sequence {
   public static void main(String[] args) {
-    Scanner input = new Scanner(System.in);
-    System.out.println("Enter an INTEGER sequence(use spaces(If you have finished,press ENTER)):");
-    int currentElement = 0;
-    int previousElement = 0;
+    Sequence sequence = new Sequence();
+    Checker checker = new Checker();
     try {
-      previousElement = input.nextInt();
-    } catch (Exception e) {
-      System.out.println("It's NOT integer value. Please try again.");
-      return;
-    }
-    while (input.hasNextInt()) {
-      currentElement = input.nextInt();
-      if (currentElement < previousElement) {
-        System.out.println("It is NOT non-decreasing sequence.");
-        return;
-      } else {
-        previousElement = currentElement;
+      if (args.length == 0) {
+        ConsoleReader consoleReader = new ConsoleReader();
+        sequence.printResultOfCheck(checker.check(consoleReader.getSequence()));
+      } else { 
+        sequence.printResultOfCheck(checker.check(args));  
       }
+    } catch (Exception a) {
+      System.out.println("It is not a numbers.Please try again.");
```

Comment:
try again what? application just finishes right after that message and doesn't allow to try again anything